File: 2023/python/11.py
with open("../test-cases/inp1.txt", "r") as inpfile:
    sumofcalibval = 0
    
    for line in inpfile:
        digits = []
        
        for c in line:
            if c.isdigit():
                digits.append(c)
        
        sumofcalibval += int(digits[0] + digits[-1])

print("Sum of Calibaration Value is: " + str(sumofcalibval))
inpfile.close()

File: 2023/cpp/11.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void)
{
    ifstream inpfile;
    inpfile.open("../test-cases/inp1.txt");

    string line;
    int sumofcalibval = 0;

    while (getline(inpfile, line)) {
        string digits, calibVal;

        for (const char c : line)
            if (isdigit(c)) digits += c;
        
        calibVal = calibVal + digits[0] + digits.back();
        sumofcalibval += stoi(calibVal);
    }

    cout << "Sum of calibaration value: " << sumofcalibval << endl;

    inpfile.close();
    return 0;
}